### PR TITLE
Debug bounding box model with minimal baseline

### PR DESCRIPTION
**Pull request by claust**
Create minimal test setup to verify learning fundamentals:
- dataset.py: On-the-fly 64x64 images with colored squares
- model.py: Tiny CNN (~24K params) for center prediction
- train.py: Training loop with verification checks
- visualize.py: Box visualization helpers

Results: Overfit tests PASS, full training shows learning but needs tuning to reach target loss < 0.01

### CODE REVIEW

File: baseline_sanity/visualize.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+    # Draw cross
+    for dx in range(-marker_size, marker_size + 1):
+        x = px + dx
+        if 0 <= x < w:
+            img[py, x] = color
+    for dy in range(-marker_size, marker_size + 1):
+        y = py + dy
+        if 0 <= y < h:
+            img[y, px] = color
```
[the remark this answers is not in the document]
Comment (claust) [reply baseline_sanity/visualize.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/claust/pussel/pull/13#discussion_r2634801357)
